bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/airqHandler.java

```diff
@@ -0,0 +1,956 @@
+                // pollingJob.notify();
+                // TODO: handle data refresh
+            }
+        }
+
+        // TODO: handle command
+
+        // Note: if communication with thing fails for some reason,
+        // indicate that by setting the status with detail information:
+        // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+        // "Could not control device at IP address x.x.x.x");
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("air-Q - airqHandler - initialize(): ipaddress={}, password={}",
+                getThing().getConfiguration().get("ipAddress"), getThing().getConfiguration().get("password"));
+        // set the thing status to UNKNOWN temporarily and let the background task decide for the real status.
+        // the framework is then able to reuse the resources from the thing handler initialization.
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.UNKNOWN);
+        if (getThing().getConfiguration().get("ipAddress") != null) {
+            ipaddress = getThing().getConfiguration().get("ipAddress").toString();
+        }
+        if (getThing().getConfiguration().get("password") != null) {
+            password = getThing().getConfiguration().get("password").toString();
+        }
+        if ((ipaddress.equals("")) || (password.equals(""))) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "IP Address and the device password must be provided to access your air-Q.");
+            return;
+        } else {
+            // we try if the device is reachable and the password is correct. Otherwise a corresponding message is
+            // thrown in Thing manager.
+            String data = getDecryptedContentString("http://".concat(ipaddress.concat("/data")), "GET", null);
+            if (data == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "We tried to get data from the air-Q device, but failed. Maybe the password is wrong.");
+            }
+        }
+
+        scheduler.execute(() ->
+
+        {
+            boolean thingReachable = true; // <background task with long running initialization here>
+            // when done do:
+            if (thingReachable) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+        });
+
+        // The following code will be called regularly. We only have it here to test the function
+        // Gson code based on https://riptutorial.com/de/gson
+        Runnable pollData = new Runnable() {
+
+            @Override
+            public void run() {
+                logger.trace("air-Q - airqHandler - run(): starting polled data handler");
+                if ((!ipaddress.equals("")) && (!password.equals(""))) {
```

Comment:
This I believe is more efficient as it does not compare two strings. Please change any other code to use this.

```suggestion
                if (!ipaddress.isEmpty() && !password.isEmpty()) {
```